### DISCUSSION TITLE
Run tests for all projects in a certain folder

### DISCUSSION THREAD

**Discussion by swiatek25** [sbt/sbt#7370] [Q&A] [answered]
Hi, I'm trying to define a task to run tests for all projects that are defined in a specific folder. I do not want to list all of them explicitly in the aggregate test task (list is rather long > 100 projects) but rather have them dynamically listed and places in the task definition. 

My current approach: 

```scala
lazy val pipelineApps = file("pipeline-app").listFiles
  .filter(_.isDirectory)
  .filter { projectDirectory =>
    val versionSbtFile = new File(projectDirectory, "version.sbt")
    versionSbtFile.exists() && versionSbtFile.isFile
  }
  .map(_.getName)
  .toSeq

runPipelineAppTests :=
  Def.sequential(
    pipelineApps.map(projectName => (test in LocalProject(projectName)))
  ).value
```

Unfortunately I get:
```shell
Reference to undefined setting: 
[error] 
[error]   ppi / test from runPipelineAppTests (/home/user/build.sbt:1745)
[error]      Did you mean ppi / assemblyPackageScala / test ?
```
When running `ppi/test` from shell all works fine - test task is available. 

I'm I missing something, is that even possible?

Thanks,

**Comment by adpi2** [comment, marked as answer]
> When running ppi/test from shell all works fine - test task is available.

There is a fallback mechanism in the shell. When you type `ppi/test`, sbt does not find the task, so it tries `ppi/Compile/test` and then `ppi/Test/test`. 

> I'm I missing something, is that even possible?

The `test` task is defined in the `Test` config. You need to use `LocalProject(projectName) / Test / test`.
